extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -151,28 +153,29 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined,
 					continue;
 				}
 
-				const newConnectionStringValue = await vscode.window.showInputBox(
-					{
-						title: constants.enterConnectionString,
-						ignoreFocusOut: true,
-						value: 'Server=localhost;Initial Catalog={db_name};User ID=sa;Password={your_password};Persist Security Info=False',
-						validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
-					}
-				) ?? '';
-
-				if (!newConnectionStringValue) {
-					// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
-					continue;
+				// get connection string for sql server
+				let connectionString: string = '';
+				const connectionInfo = await vscodeMssqlApi.promptForConnection(true);
```

Comment:
I'm thinking it would still be good to let them manually enter in a connection string if they want. That would also help the "disconnected" scenario because they could just do that if they didn't want to use an actual active server.

Reply:
@dzsquared @chlafreniere Thoughts? It does add another quickpick the user has to go through but in general I think we should opt on the side of giving the user more control when possible. (that way they can also add in connection string stuff that we don't directly support adding through the prompt such as timeout)

Reply:
I would prefer us to allow users to paste in a connection string in the limit for sure.

We're first focusing on the "connected" mode here, but if we allow users to freely paste in a connection string, does that radically change the rest of the flow? My assumption is that, if a user passes in a connection string, then we just skip the subsequent API calls and are on our merry way.

Are there any other assumptions that are made later on where we assume that a connection profile exists?

Reply:
I agree that we can definitely allow the user to enter their own SQL Connection String. I just wanna understand the flow a bit. 
After the user enter the "connection string setting name" they are currently shown this (current version - without my edits):
![image](https://user-images.githubusercontent.com/23587151/148475472-65095a91-5df6-41b4-a75d-cd3e23cbd280.png)

Would we add a menu item to the above prompt to say "Create a SQL Connection String from SQL Connection"? Which would then go through the rest of my flow - of selecting a connection profile or adding one and then creating the SQL connection?

Reply:
You would have to have a quickpick where the user chooses to either select a connection profile or enter the connection string manually - then either display the connection prompt (for select connection profile) or the input as in the screenshot above (for manually enter)

Reply:
@Charles-Gagnon please see the updated screenshots in the description. Discussed with Chris prior to setting the text. Please let me know what you think!